src/Pipeline.h

```diff
@@ -455,6 +455,21 @@ class Pipeline {
      * been rescheduled. */
     void invalidate_cache();
 
+    /** Add a top-level precondition to the generated pipeline,
+     * expressed as a boolean Expr. The Expr may depend on parameters
+     * only, and may not call any Func or use a Var. If the condition
```

Comment:
> may not call any Func or use a Var

What happens if this is violated -- do we check and produce an error, or is it UB?

Reply:
Are the requirements checked in the order added, or is the order (potentially) arbitrary? (Should be documented either way)

Reply:
References to Func/Var are checked in Pipeline::add_requirement (produces a user_error).